src/query/get-users.query.ts

```diff
@@ -0,0 +1,54 @@
+import { verifyToken } from "../cryptography/verify-token";
+import { AppDataSource } from "../data-source";
+import { User } from "../entity/User";
+import { CustomError } from "../errors/login-error-class";
+import { isAuthorized } from "../validators/authorization-validator";
+
+export async function getUserList(
+  token: string,
+  options?: { page?: number; limit?: number }
+) {
+  let numberOfUsers: number = options?.limit ? options.limit : 10;
+  const page: number = options?.page
+    ? options.page <= 0
+      ? 1
+      : options.page
+    : 1;
+  const offset: number = (page - 1) * numberOfUsers;
+  const DataSourceLength = await AppDataSource.manager.count(User);
+  const hasPreviousPage: boolean = page != 1;
```

Comment:
```suggestion
  const hasPreviousPage: boolean = page !== 1;
```

Eu costumo recomendar sempre usar comparação estrita.